lib/src/mount.rs

```diff
@@ -50,13 +58,42 @@ fn run_findmnt(args: &[&str], path: &str) -> Result<Filesystem> {
 /// Inspect a target which must be a mountpoint root - it is an error
 /// if the target is not the mount root.
 pub(crate) fn inspect_filesystem(path: &Utf8Path) -> Result<Filesystem> {
-    run_findmnt(&["--mountpoint"], path.as_str())
+    findmnt_filesystem(&["--mountpoint"], path.as_str())
 }
 
 #[context("Inspecting filesystem by UUID {uuid}")]
 /// Inspect a filesystem by partition UUID
 pub(crate) fn inspect_filesystem_by_uuid(uuid: &str) -> Result<Filesystem> {
-    run_findmnt(&["--source"], &(format!("UUID={uuid}")))
+    findmnt_filesystem(&["--source"], &(format!("UUID={uuid}")))
+}
+
+// Check if a specified device contains an already mounted filesystem
+pub(crate) fn is_mounted_in_pid1_mountns(path: String) -> Result<bool> {
+    let o = run_findmnt(&["-N"], "1")?;
+    for fs in o.filesystems.iter() {
+        if is_source_mounted(&path, fs)? {
+            return Ok(true);
+        }
+    }
+
+    Ok(false)
+}
+
+// Recursively check a given filesystem to see if it contains an already mounted source
+pub(crate) fn is_source_mounted(path: &String, mounted_fs: &Filesystem) -> Result<bool> {
+    if mounted_fs.source.contains(path) {
```

Comment:
A bit uncertain about `source.contains(path)` vs `mounted_fs.source.as_str() == path` here...I guess you're probably doing that as a way to handle e.g. `/dev/vda3` being mounted, but we're trying to match vs a whole device like `/dev/vda`?

Just thinking through this...what is probably cleaner actually is to do a set-intersection; on the [calling side](https://github.com/containers/bootc/pull/805/files#diff-e44cbc45f98e122e00481bbcf78595a5c93935f6ef5bd0e4dd32c6880f1f2cfaR167) loop through all children of the device.

Reply:
I'm not sure I follow what you're saying, my understanding is that we want to see if `/dev/vda` (the device) is mounted or if any partition on it is mounted. That's what I wrote it for, so it will kickback the mounting error if any part of the device is mounted. 

Are you saying that this should only fail if the device alone is mounted? As in it shouldn't be checking the partitions of the device?

That's the reasoning behind the `source.contains(path)`, I can't be sure what number partition will be mounted but I can work off it having the same naming stem.

Reply:
Or maybe cleaner actually, make `is_mounted_in_pid1_mountns` take a `&Device` and have it iterate over children.

Reply:
So instead of looping through the `findmnt` output and searching for a matching device, loop through the device's children and check each to see if it's mounted?

Reply:
That was my original thought a couple of weeks ago when I first looked at this, I ended up going with parsing the `findmnt` output because it seemed more straightforward for reasons I can't exactly remember now.

 For the code block you noted above, should mount checking be within the `opts.wipe` handling? Or are you just using that as an example of what the loop would look like?

Reply:
Eh nevermind, no need to bikeshed this one fully, I think it's fine as is. Let's just do the minor tweaks and get it in!

Reply:
Copy that, will push as soon as I run through my testing